watchtower-plugin/src/main.rs

```diff
@@ -308,8 +337,12 @@ async fn retry_tower(
 ) -> Result<serde_json::Value, Error> {
     let tower_id = TowerId::try_from(v).map_err(|e| anyhow!(e))?;
     let state = plugin.state().lock().unwrap();
-    if let Some(tower_status) = state.get_tower_status(&tower_id) {
-        if let Some(retrier_status) = state.retriers.get(&tower_id) {
+    if let Some(tower) = state.towers.get(&tower_id) {
+        if !should_send(tower.net_addr.addr_type(), state.always_use_proxy) {
+            return Err(anyhow!(
+                "Cannot retry a clearnet tower when running in always-use-proxy mode"
+            ));
```

Comment:
You know, this error message got me thinking, why not send the request to the clearnet tower using the proxy anyways? Are we assuming it is "just a proxy" or "a Tor proxy specifically"?

Reply:
I just ran `torsocks curl -i http://vpn.mariocynicys.cf/ip` and it gave me an IP which is indeed not mine.

Reply:
It is, indeed, a Tor only proxy. We are following the `proxy` / `always-use-proxy` logic from CLN:

https://github.com/ElementsProject/lightning/blob/master/doc/TOR.md

Reply:
Ok fine, but I have read this part in the doc you linked from cln:
```
4. always-use-proxy informs Core Lightning to always use Tor even when connecting to nodes with public IPs. You can set this to false or remove it, if you are not privacy-conscious and find Tor is too slow for you.
```
and I interpreted it as: always use proxy even with public IP (clearnet?) nodes,
and not: don't ever try to contact a public IP node.

Reply:
Ummm, that may be the case indeed, in which case the logic could be heavily simplified.

Let me ask to the CLN devs because I may have gotten that wrong in the first place.

Reply:
I've just chatted with @cdecker about this and you're right. It refers to sending **through** Tor.

I'll update the functionality accordingly (and most likely simplify this quite a bit)